handlers/attachment_resource.go

```diff
@@ -38,7 +38,13 @@ func HandleSubmitAttachment(svc dao.Service) http.Handler {
 		}
 
 		// Validate that the provided attachment details are correct
-		validationErrs := service.ValidateAttachmentDetails(attachmentType, header)
+		validationErrs, err := service.ValidateAttachmentDetails(svc, transactionID, attachmentType, header)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("error retrieving attachments from db: [%s]", err))
```

Comment:
Might not be a DB error
```suggestion
			log.ErrorR(req, fmt.Errorf("error validating attachment details: [%s]", err))
```